main.tf

```diff
@@ -322,14 +322,16 @@ module "cache" {
 ### Trust policy
 ################################################################################
 resource "aws_iam_instance_profile" "instance" {
-  name = "${local.name_iam_objects}-instance"
+  count = var.create_runner_iam_role ? 1 : 0
+
+  name = local.aws_iam_role_instance_name
   role = local.aws_iam_role_instance_name
   tags = local.tags
```

Comment:
Make name customisable based on runner_iam_role_name variable.
Prevent creating instance profile if a custom external role is used - in that case all resources relating to the role should be created and the module should just use them.